src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -0,0 +1,1062 @@
+                        {
+                            foreach (int p in cookie.PortList)
+                            {
+                                if (p == port)
+                                {
+                                    to_add = true;
+                                    break;
+                                }
+                            }
+                        }
+                        else
+                        {
+                            // It was implicit Port, always OK to add.
+                            to_add = true;
+                        }
+
+                        // Refuse to add a secure cookie into an 'unsecure' destination
+                        if (cookie.Secure && !isSecure)
+                        {
+                            to_add = false;
+                        }
+
+                        if (to_add)
+                        {
+                            // In 'source' are already orederd.
+                            // If two same cookies come from different 'source' then they
+                            // will follow (not replace) each other.
+                            destination.InternalAdd(cookie, false);
+                        }
+                    }
+                }
+            }
+        }
+
+        public string GetCookieHeader(Uri uri)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException("uri");
+            }
+            string dummy;
+            return GetCookieHeader(uri, out dummy);
+        }
+
+        internal string GetCookieHeader(Uri uri, out string optCookie2)
+        {
+            CookieCollection cookies = InternalGetCookies(uri);
+            string cookieString = String.Empty;
+            string delimiter = String.Empty;
+
+            foreach (Cookie cookie in cookies)
+            {
+                cookieString += delimiter + cookie.ToString();
+                delimiter = "; ";
+            }
+            optCookie2 = cookies.IsOtherVersionSeen ?
+                          (Cookie.SpecialAttributeLiteral +
+                           Cookie.VersionAttributeName +
+                           Cookie.EqualsLiteral +
+                           Cookie.MaxSupportedVersion.ToString(NumberFormatInfo.InvariantInfo)) : String.Empty;
+
+            return cookieString;
+        }
+
+        public void SetCookies(Uri uri, string cookieHeader)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException("uri");
+            }
+            if (cookieHeader == null)
+            {
+                throw new ArgumentNullException("cookieHeader");
+            }
+            CookieCutter(uri, null, cookieHeader, true); // Will throw on error
+        }
+    }
+
+    internal class PathList
+    {
+        private SortedList _list = (SortedList.Synchronized(new SortedList(PathListComparer.StaticInstance)));
```

Comment:
I'll make this `readonly`, but I'm going to leave the collection type the same for the time being.